flask/app/routes.py

```diff
@@ -340,3 +355,174 @@ def get_enums():
         if issubclass(obj, Enum) and name != 'Enum':
             enums[name] = [e.value for e in getattr(models, name)]
     return jsonify(enums)
+
+
+def enum_validate(entity: db.Model, json_mixin: Dict[str, any], columns: List[str]) -> Union[None, str]:
+
+    for field in columns: 
+        if field in json_mixin:
+            column = getattr(entity, field) # the column type from the entities
+            value = json_mixin[field]
+            if hasattr(column.type, 'enums'): # check if enum
+                if value not in column.type.enums:
+                    allowed = column.type.enums
+                    return f'Invalid value for: "{field}", current input is "{value}" but must be one of {allowed}'
+
+from io import StringIO
+@app.route('/api/_bulk', methods = ['POST'])
+@login_required
+def bulk_update():
+
+    dataset_ids = [] 
+
+    editable_dict = {'participant' : ['participant_codename', 'sex', 'participant_type',
+                                        'month_of_birth', 'affected', 'solved', 'notes'],
+                        'dataset' : ['dataset_type', 'input_hpf_path', 'notes', 'condition',
+                                    'extraction_protocol', 'capture_kit', 'library_prep_method',
+                                    'library_prep_date', 'read_length', 'read_type', 'sequencing_id',
+                                    'sequencing_date', 'sequencing_centre', 'batch_id', 'discriminator' ],
+                    'tissue_sample': ['extraction_date', 'tissue_sample_type','tissue_processing', 'notes']}
+
+    if 'csv' not in request.headers['Content-Type']:
+        return "Only Content Type 'text/csv' Supported", 415
+    else:
+        try:
+            dat = pd.read_csv(StringIO(request.data.decode("utf-8")))
+            dat = dat.where(pd.notnull(dat), None)
+        except Exception as err:
+            raise err
```

Comment:
If we're not doing anything with the exception then there's no need for a try-except.